src/Compilers/CSharp/Test/Symbol/Symbols/RequiredMembersTests.cs

```diff
@@ -5820,6 +5820,462 @@ public Derived() : base() { }
+            //     public DerivedDerived(string str) : base()
+            Diagnostic(ErrorCode.WRN_UninitializedNonNullableField, "DerivedDerived").WithArguments("property", "Str").WithLocation(25, 12),
+            // (25,12): warning CS8618: Non-nullable property 'Str' must contain a non-null value when exiting constructor. Consider adding the 'required' modifier or declaring the property as nullable.
+            //     public DerivedDerived(string str) : base()
+            Diagnostic(ErrorCode.WRN_UninitializedNonNullableField, "DerivedDerived").WithArguments("property", "Str").WithLocation(25, 12)
+        );
+    }
+
+    [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/74423")]
+    public void SetsRequiredMembersHonoredForPropertyOverride_07()
+    {
+        var code = """
+            #nullable enable
+            using System.Diagnostics.CodeAnalysis;
+
+            public abstract class Base
+            {
+                public Base()
+                {
+                }
+    
+                public required abstract string Str { get; set; }
+            }
+
+            public class Derived : Base
+            {
+                public Derived() : base()
+                {
+                }
+
+                public override required string Str { get; set; }
+            }
+
+            public class DerivedDerived : Derived
+            {
+                [SetsRequiredMembers]
+                public DerivedDerived(string str) : base()
+                {
+                }
+            }
+            """;
+
+        var comp = CreateCompilationWithRequiredMembers(code);
+        comp.VerifyDiagnostics(
+            // (25,12): warning CS8618: Non-nullable property 'Str' must contain a non-null value when exiting constructor. Consider adding the 'required' modifier or declaring the property as nullable.
+            //     public DerivedDerived(string str) : base()
+            Diagnostic(ErrorCode.WRN_UninitializedNonNullableField, "DerivedDerived").WithArguments("property", "Str").WithLocation(25, 12)
+        );
+    }
+
+    [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/74423")]
+    public void SetsRequiredMembersHonoredForPropertyOverride_08()
+    {
+        var code = """
+            #nullable enable
+            using System.Diagnostics.CodeAnalysis;
+
+            public abstract class Base
+            {
+                [SetsRequiredMembers]
+                public Base(string str)
+                {
+                    this.Str = str;
+                }
+    
+                public required abstract string? Str { get; set; }
+            }
+
+            public class Derived : Base
+            {
+                [SetsRequiredMembers]
+                public Derived(string str) : base(str)
+                {
+                }
+
+                public override required string Str { get; set; }
```

Comment:
We should add a test which is similar to this except with `[AllowNull]` on this override. This may actually have the effect of removing both warnings.